services/transformer/handler.go

```diff
@@ -0,0 +1,118 @@
+package transformer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type transformerHandler struct {
+	logger    logger.Logger
+	asyncInit *misc.AsyncInit
```

Comment:
we don't need an `AsyncInit` here, just a `chan struct{}` since we don't need to wait for multiple asynchronous tasks to finish, just one

Reply:
Used channel